### DISCUSSION TITLE
After update to 6.1.19 (importer 1.5.6) Importer can not makes all transactions deposits no matter roles

### DISCUSSION THREAD

**Discussion by Nicholas-Cochrane** [firefly-iii/firefly-iii#9308] [Firefly III]
### Support guidelines

- [x] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [x] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [x] ... there are no open or closed issues that are related to my problem
- [x] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

When importing CSVs in a format that previously worked, all transactions are deposits.

### Debug information

Debug information generated at 2024-10-04 13:07:21 for Firefly III version **v6.1.19**.

<table><thead><tr><th colspan="2">System information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Firefly III</td><td>6.1.19 / v2.1.0 / 24 (exp. 24)</td></tr><tr><td>PHP version</td><td>8.3.9 (64bits) / apache2handler / Linux x86_64</td></tr><tr><td>BCscale</td><td>12</td></tr><tr><td>Error reporting</td><td>Display: Off, reporting: ALL errors</td></tr><tr><td>Max upload</td><td>67108864 (64 MB)</td></tr><tr><td>Database drivers</td><td><strong>*mysql*</strong>, pgsql, sqlite, </td></tr><tr><td>Docker build</td><td><span>#</span>1113, base <span>#</span>86</td></tr></tbody></table><table><thead><tr><th colspan="2">Firefly III information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Timezone</td><td>America/Los_Angeles + America/Los_Angeles</td></tr><tr><td>App environment</td><td>local, debug: false</td></tr><tr><td>Layout</td><td>v1</td></tr><tr><td>Logging</td><td>notice, stack / (empty)</td></tr><tr><td>Cache driver</td><td>file</td></tr><tr><td>Default language and locale</td><td>en_US + equal</td></tr><tr><td>Trusted proxies</td><td>*</td></tr><tr><td>Login provider & user guard</td><td>eloquent / web</td></tr><tr><td>Login headers</td><td>N/A + N/A</td></tr><tr><td>Stateful domains</td><td></td></tr><tr><td>Last cron job</td><td>2024-10-03 07:00:00 (1 day ago)</td></tr><tr><td>Mailer</td><td>log</td></tr></tbody></table><table><thead><tr><th colspan="2">User-specific information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>User</td><td><span>#</span>1 of 1</td></tr><tr><td>User flags</td><td><span title="Has rules">:wrench:</span></td></tr><tr><td>Session start</td><td>2024-07-01 12:10:07</td></tr><tr><td>Session end</td><td>2024-10-04 12:10:07</td></tr><tr><td>View range</td><td>1M</td></tr><tr><td>User language</td><td>en_US</td></tr><tr><td>User locale</td><td>en_US</td></tr><tr><td>Locale(s) supported</td><td>en_US.utf8: :white_check_mark:<br>en_US.UTF-8: :white_check_mark:<br></td></tr><tr><td>User agent</td><td>Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:131.0) Gecko/20100101 Firefox/131.0</td></tr></tbody></table>

### Expected behaviour

Import The following data as 2 deposits and one withdrawal 

date  | type.......| description | debit | credit
x/x/x | CREDIT | .......................|..............| 28.40
x/x/x | DEBIT ...| ........................| -111...|............
x/x/x | CREDIT | .......................| .............| 14.53

Ignore the '.......' I just wanted to format it and it was deleting blank space and marking '_' as mark down and I am too frustrated to find a better way

Date -> Transaction date
Type -> Bank debit/credit indicator (also tried ignore this column)
Description -> Description 
Debit -> Amount (debit column) (also tried amount, amount (negated column))
Credit -> Amount (credit column) (also tried setting both to amount)

The previous version (which unfortunately I don't know what it was but the json file shows the imported version to be 1.4.4) worked fine with no issues.

### Steps to reproduce

import with above settings
the DEBIT transaction becomes a deposit of (111 USD)

The full json of the import configuration is below, though as stated above I tried various modifications and not importing the json file and doing everything by hand to see if that was the issue. And after the first import duplicate detection had to be disabled.
`{
    "version": 3,
    "source": "ff3-importer-1.4.4",
    "created_at": "2024-02-17T21:11:29-08:00",
    "date": "n\/j\/Y",
    "default_account": 1,
    "delimiter": "comma",
    "headers": true,
    "rules": true,
    "skip_form": false,
    "add_import_tag": true,
    "roles": [
        "date_transaction",
        "internal_reference",
        "generic-debit-credit",
        "description",
        "amount_debit",
        "amount_credit",
        "external-id",
        "note"
    ],
    "do_mapping": [
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false
    ],
    "mapping": [],
    "duplicate_detection_method": "classic",
    "ignore_duplicate_lines": true,
    "unique_column_index": 0,
    "unique_column_type": "internal_reference",
    "flow": "file",
    "content_type": "csv",
    "custom_tag": "",
    "identifier": "0",
    "connection": "0",
    "ignore_spectre_categories": false,
    "grouped_transaction_handling": "",
    "use_entire_opposing_address": false,
    "map_all_data": false,
    "accounts": [],
    "date_range": "",
    "date_range_number": 30,
    "date_range_unit": "d",
    "date_not_before": "",
    "date_not_after": "",
    "nordigen_country": "",
    "nordigen_bank": "",
    "nordigen_requisitions": [],
    "nordigen_max_days": "90",
    "conversion": false,
    "ignore_duplicate_transactions": true
}`

### Additional info

Also, is there any way to remove duplicate hashes from the duplicate detection? it's annoying to have to not have duplicate detection when you delete a bunch of incorrectly imported transactions. I have had this issues before too, though last time it was user error that caused the transactions to be imported wrong and I had to remove them and import them agian.

**Comment by JC5** [comment]
Hey, thanks for opening an issue. I would like to start by getting some real example lines, could you share some? Feel free to censor data of course.

> Also, is there any way to remove duplicate hashes from the duplicate detection?

https://docs.firefly-iii.org/how-to/data-importer/import/duplicates/#deleted-but-duplicate

**Comment by Nicholas-Cochrane** [comment]
I was making some examples which I will still put in here but this time 'amount (negated column)' worked and I don't know why it didn't last time. That said the initial configuration was working in my previous build so I am still not sure as to why it would have changed.
Example CSV
[EXAMPLE.csv](https://github.com/user-attachments/files/17267005/EXAMPLE.csv)


Example 1:
[import_config_2024-10-05 Example 1.json](https://github.com/user-attachments/files/17267008/import_config_2024-10-05.Example.1.json)
![Example 1 Importer](https://github.com/user-attachments/assets/fd990dbc-6bf4-49c0-8b08-a223e3f6375d)
![Example 1 Firefly](https://github.com/user-attachments/assets/a2a399a1-5859-48d3-ae96-29a11f74bbef)


Example 2: Negate column and rules disabled (I also tried with rules enabled and it also was working again)
I am going to try importing my data and see if it works I will report back if I encounter issues again.
[import_config_2024-10-05 Example 2.json](https://github.com/user-attachments/files/17267011/import_config_2024-10-05.Example.2.json)
![Example 2 Importer](https://github.com/user-attachments/assets/c9d29191-32f9-4e07-ab15-c76b8f06eef2)
![Example 2 Firefly](https://github.com/user-attachments/assets/b854b818-2c7e-4494-99b0-76df258db5f8)

**Comment by Nicholas-Cochrane** [comment]
The issues does persist with credit cards 
EXAMPLE CSV
[EXAMPLE Credit.csv](https://github.com/user-attachments/files/17267123/EXAMPLE.Credit.csv)


Example 1:
[import_config_2024-10-05 Credit Example 1.json](https://github.com/user-attachments/files/17267126/import_config_2024-10-05.Credit.Example.1.json)
![Credit Example 1](https://github.com/user-attachments/assets/6c7e8ae3-5df4-459a-9c91-d5d1dd0118f6)
![Credit Example 1 Firefly](https://github.com/user-attachments/assets/062e6dc6-78b9-42f0-bc2d-bf1c2798dd55)

Example 2:
[import_config_2024-10-05 Credit Example 2.json](https://github.com/user-attachments/files/17267127/import_config_2024-10-05.Credit.Example.2.json)
![Credit Example 2](https://github.com/user-attachments/assets/de011ca5-159f-4601-a887-89abe0590dc5)
![Credit Example 2 Firefly](https://github.com/user-attachments/assets/0a8444f9-175d-4a3d-8379-568a7049dc6d)

Expected Result:
![Credit Expected](https://github.com/user-attachments/assets/3a78ad73-7483-4b80-8ce7-09c16c34965a)

**Reply by JC5** [reply]
It looks like you use the wrong column type for the `Debit` column. This should be `amount_debet`, NOT `amount_negated`.

**Reply by Nicholas-Cochrane** [reply]
Credit Example 2 is using amount_debit.
